Subject: hv: reduce storvsc_ringbuffer_size from 1M to 128K to simplify booting with 1k vcpus
From: <ohering@suse.de>
Patch-mainline: Never, this change carry until upstream provides better method
References: fate#323887

This is actually related to bsc#1122822. Michael Kelley says:
The current size of 1 Mbyte is much bigger than is needed. Experiments
about a year ago showed that 128 Kbytes is quite sufficient, and I never
got around to changing the default size in the upstream code. I'll try
to push the upstream change in the next few days.

Workaround so far is:
boot up  with "hv_storvsc.storvsc_ringbuffer_size=131072"

Acked-by: Olaf Hering <ohering@suse.de>

--- a/drivers/scsi/storvsc_drv.c
+++ b/drivers/scsi/storvsc_drv.c
@@ -385,7 +389,7 @@ enum storvsc_request_type {
  * This is the end of Protocol specific defines.
  */
 
-static int storvsc_ringbuffer_size = (256 * PAGE_SIZE);
+static int storvsc_ringbuffer_size = (128 * 1024);
 static u32 max_outstanding_req_per_channel;
 
 static int storvsc_vcpus_per_sub_channel = 4;
